pennylane_lightning/core/src/algorithms/AdjointJacobianBase.hpp

```diff
@@ -59,7 +59,8 @@ template <class StateVectorT, class Derived> class AdjointJacobianBase {
             state.applyOperation(operations.getOpsName()[op_idx],
                                  operations.getOpsWires()[op_idx],
                                  operations.getOpsInverses()[op_idx] ^ adj,
-                                 operations.getOpsParams()[op_idx]);
+                                 operations.getOpsParams()[op_idx],
+                                 operations.getOpsMatrices()[op_idx]);
```

Comment:
Do we want to always dispatch to the maxtrix-backed apply? What if the matrix is empty, but still provided?

Reply:
L-GPU and L-Kokkos already had an optional `matrix` argument in `applyOperation` which defaults to an empty vector. For L-Qubit, if the dynamic dispatcher hits a named kernel it will ignore `matrix`, otherwise it will pass it to the matrixop dispatcher.